code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -1092,6 +1092,10 @@ var/global/list/damage_icon_parts = list()
 		O.name = "[index]"
 		O.icon = t_inhand_state
 		O.icon_state = t_state
+		if(I.color)
+			O.color = I.color
+		else
+			O.color = null // this prevents a bug where all inhands would be colored after holding a colored item
```

Comment:
I mean
Can't you just say `O.color = I.color` without the if/else
Because if it's null it gets set to null
And if it's not null it gets set to the color